csharp-tree-sitter.el

```diff
@@ -27,10 +26,13 @@
 ;;; Code:
 (require 'cl-lib)
 (require 'seq)
+
 (require 'tree-sitter)
 (require 'tree-sitter-hl)
 (require 'tree-sitter-indent)
 
+(require 'csharp-mode)
```

Comment:
Why do we have to require `csharp-mode` here? Not sure I follow this change. I tried to keep them as separate as possible, but it might not work as I intend it to. Can you expand a little on this?

Reply:
I was trying to avoid the following warnings. Of course, it just warnings so I can remove it if the team don't mind about this! ;)

```
In csharp-tree-sitter-mode:
csharp-tree-sitter.el:270:9:Warning: assignment to free variable
    ‘csharp-mode-syntax-table’
csharp-tree-sitter.el:271:9:Warning: assignment to free variable
    ‘csharp-mode-map’
```

Reply:
Right! Can you try two things for me?

1. `(defvar cshaerp-mode-syntax-table)` and `csharp-mode-map` somewhere inside `csharp-tree-sitter.el` to forward declare them, thus making them known by the mode?
2. Remove them altogether from `csharp-tree-sitter.el` and checking if that breaks a lot of stuff.

I think we may not even need them anymore, as emacs sets some defaults when they are not provided, and I don't remember if I put them in before or after I split it to two separate mojor modes.

I think we should avoid warnings, so we need to fix this some way :) I just didn't get them on my end. Not sure why.

Reply:
Okay, I have switched from requiring the whole module csharp to forward declaration.